drivers/spi/spi_nrfx_spim.c

```diff
@@ -568,7 +568,7 @@ static int spim_nrfx_pm_action(const struct device *dev,
 	static const struct spi_nrfx_config spi_##idx##z_config = {	       \
 		.spim = NRFX_SPIM_INSTANCE(idx),			       \
 		.max_chunk_len = (1 << SPIM##idx##_EASYDMA_MAXCNT_SIZE) - 1,   \
-		.max_freq = SPIM##idx##_MAX_DATARATE * 1000000,		       \
+		.max_freq = SPIM_PROP(idx, max_frequency),		       \
```

Comment:
That's fine, but perhaps we need some additional checks to make sure the user has not changed this value and that future SoC ports do not make any mistakes here.

I suggest adding some checks in soc/arm/nordic_nrf/, analogous to the validate_enabled_instances.c and validate_base_addresses.c files in that directory.